src/core_plugins/kibana/public/management/sections/indices/_edit.js

```diff
@@ -70,23 +71,37 @@ uiModules.get('apps/management')
   });
 
   $scope.refreshFields = function () {
-    $scope.indexPattern.refreshFields();
+    const confirmModalOptions = {
+      confirmButtonText: 'Reload fields',
+      onConfirm: () => { $scope.indexPattern.refreshFields(); }
+    };
+    confirmModal(
+      'This will reset the field popularity counters. Are you sure you want to reload your fields?',
```

Comment:
The tooltip on the icon for this operation says "Refresh fields". For consistency we should either change that to "Reload fields" or change this confirmation string text to "... refresh your fields?".